sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -40,10 +42,16 @@ private[sql] class SharedState(val sparkContext: SparkContext) {
    */
   val listener: SQLListener = createListenerAndUI(sparkContext)
 
+  lazy val hadoopConf: Configuration = {
+    val conf = sparkContext.hadoopConfiguration
+    conf.addResource(Utils.getContextOrSparkClassLoader.getResource("hive-site.xml"))
```

Comment:
is it better than `conf.addResouorce("hive-site.xml")`? Which corner case do we worry about it? cc @yhuai